### PR TITLE
Accept numbers in place of strings in JSON

### PR DESCRIPTION
**Pull request by tobim**
###  :notebook_with_decorative_cover: Description

If the schema says string but the data is numerical, we call `to_string` instead of failing.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.

### CODE REVIEW

File: libvast/test/format/json.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
  *                                                                            *
  * This file is part of VAST. It is subject to the license terms in the       *
  * LICENSE file found in the top-level directory of this distribution and at  *
- * http://vast.io/license. No part of VAST, including this file, may be       *
```
Comment (dominiklohmann) [inline libvast/test/format/json.cpp:9]: Please revert this change

**Review by dominiklohmann — APPROVED** [review]
LGTM, but please address my one comment before merging.

File: libvast/test/format/json.cpp (hunk 1 above)
Comment (tobim) [reply libvast/test/format/json.cpp:9]: That was unintentional.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ This changelog documents all notable user-facing changes of VAST.
 
 ## Unreleased
 
+- 🐞 The JSON parser now accepts data with numerical or boolean values in
```
Comment (mavam) [inline CHANGELOG.md:16]: I would make this a change instead of bug. Our parser behaved correctly beforehand, but now it operates in a "enhanced" mode and accepts inputs more liberally.

Comment (dominiklohmann) [reply CHANGELOG.md:16]: I disagree. From a users perspective, this is a bug. We do all kinds of conversions here and this one was missing.

Comment (mavam) [reply CHANGELOG.md:16]: Okay, I have not a particularly strong opinion here.
